ampd/src/broadcast/msg_queue.rs

```diff
@@ -345,17 +356,23 @@ impl Queue {
     }
 
     #[instrument(skip(handle_error))]
-    pub fn push_or<F>(&mut self, msg: QueueMsg, handle_error: F) -> Option<nonempty::Vec<QueueMsg>>
+    pub fn push_or<F>(
+        &mut self,
+        msg: QueueMsg,
+        handle_error: F,
+        monitoring_client: &monitoring::Client,
```

Comment:
there is probably no way around adding the monitoring client to MsgQueue, but I definitely don't want to add it here. you can use a closure inside MsgQueue to create a handle_error function that already closes over the monitoring client, so the Queue doesn't have to know about it